benchmark_runner/common/clouds/BareMetal/bare_metal_operations.py

```diff
@@ -0,0 +1,284 @@
+        self._provision_port = int(self._environment_variables_dict.get('provision_port', ''))
+        self._provision_timeout = int(self._environment_variables_dict.get('provision_timeout', ''))
+        self._container_private_key_path = self._environment_variables_dict.get('container_private_key_path', '')
+        self._connection_data = ConnectionData(host_name=self._provision_ip,
+                                                user_name=self._user,
+                                                port=self._provision_port,
+                                                timeout=self._provision_timeout,
+                                                ssh_key=self._container_private_key_path)
+        self._remote_ssh = RemoteSsh(self._connection_data)
+        if self._environment_variables_dict.get('github_repository_short', ''):
+            self._github_operations = GitHubOperations()
+        self._ssh = SSH()
+        self._cli = self._environment_variables_dict.get('cli', '')
+
+    def _get_kubeadmin_password(self):
+        """
+        This method returns kubeadmin password if exist
+        :return:
+        """
+        if self._remote_ssh.exist(remote_path=self._provision_kubeadmin_password_path):
+            return self._remote_ssh.run_command(f'cat {self._provision_kubeadmin_password_path}')
+
+    def _get_kubeconfig(self):
+        """
+        This method returns kubeconfig if exist
+        :return:
+        """
+        if self._remote_ssh.exist(remote_path=self._provision_kubeconfig_path):
+            return self._remote_ssh.run_command(f'cat {self._provision_kubeconfig_path}')
+
+    def _wait_for_install_complete(self, sleep_time: int = SHORT_TIMEOUT):
+        """
+        This method waits till ocp install complete
+        :param sleep_time:
+        :return:
+        """
+        current_wait_time = 0
+        while self._provision_timeout <= 0 or current_wait_time <= self._provision_timeout:
+            install_log = self._remote_ssh.run_command(self._provision_installer_log)
+            if 'failed=0' in install_log:
+                return True
+            elif 'failed=1' in install_log:
+                return False
+            logger.info(f'Waiting till OCP install complete, waiting {int(current_wait_time/60)} minutes')
+            # sleep for x seconds
+            time.sleep(sleep_time)
+            current_wait_time += sleep_time
+        raise OCPInstallationFailed(self._provision_installer_log)
+
+    def _install_ocp_cmd(self):
+        """
+        This method returns local ocp installer cmd if exist
+        :return:
+        """
+        if self._remote_ssh.exist(remote_path=self._provision_installer_path):
+            return self._provision_installer_cmd
+
+    @logger_time_stamp
+    def connect(self):
+        """
+        This method connect to remote provision machine
+        :return:
+        """
+        self._remote_ssh.connect()
+
+    @logger_time_stamp
+    def disconnect(self):
```

Comment:
Name this `disconnect_from_provisioner` or similar.

Reply:
Fixed